src/main/java/org/candlepin/subscriptions/tally/billing/ContractsController.java

```diff
@@ -64,51 +67,70 @@ public ContractsController(
               delayExpression = "#{@contractsClientProperties.getBackOffInitialInterval()}",
               maxDelayExpression = "#{@contractsClientProperties.getBackOffMaxInterval()}",
               multiplierExpression = "#{@contractsClientProperties.getBackOffMultiplier()}"))
-  public Optional<Double> getContractCoverage(BillableUsage usage) {
+  public Double getContractCoverage(BillableUsage usage) throws ContractMissingException {
+
     if (!tagProfile.isTagContractEnabled(usage.getProductId())) {
-      // Contract not enabled for product, nothing to return.
-      log.debug(
-          "Skipping contract lookup for product {} since it is not contract enabled.",
-          usage.getProductId());
-      return Optional.empty();
+      throw new IllegalStateException(
+          String.format("Product %s is not contract enabled.", usage.getProductId()));
+    }
+
+    String contractMetricId =
+        getContractMetricId(usage.getBillingProvider(), usage.getProductId(), usage.getUom());
+
+    if (ObjectUtils.isEmpty(contractMetricId)) {
+      throw new IllegalStateException(
+          String.format(
+              "Contract metric ID is not configured for provider=%s product=%s usage=%s",
```

Comment:
Instead of **usage** should it be **unit** in the message? Like unit=%s

Reply:
I changed it to uom=%s since that's what we are outputting there.